tests/test_converter.py

```diff
@@ -57,7 +57,55 @@ def test_validate_initial_user_config_expects_profile(execution_mode):
     execution_config = ExecutionConfig(execution_mode=execution_mode)
     profile_config = None
     project_config = ProjectConfig()
-    assert validate_initial_user_config(execution_config, profile_config, project_config, None) is None
+    assert validate_initial_user_config(execution_config, profile_config, project_config, None, {}) is None
+
+
+@pytest.mark.parametrize("operator_args", [{"env": {"key": "value"}}, {"vars": {"key": "value"}}])
+def test_validate_user_config_operator_args_deprecated(operator_args):
+    """Deprecating warnings should be raised when using operator_args with "vars" or "env"."""
+    project_config = ProjectConfig()
+    execution_config = ExecutionConfig()
+    render_config = RenderConfig()
+    profile_config = MagicMock()
+
+    with pytest.deprecated_call():
+        validate_initial_user_config(execution_config, profile_config, project_config, render_config, operator_args)
+
+
+@pytest.mark.parametrize("project_config_arg, operator_arg", [("dbt_vars", "vars"), ("env_vars", "env")])
+def test_validate_user_config_fails_project_config_and_operator_args_overlap(project_config_arg, operator_arg):
+    """
+    The validation should fail if a user specifies both a ProjectConfig and operator_args with dbt_vars/vars or env_vars/env
+    that overlap.
+    """
+    project_config = ProjectConfig(
+        project_name="fake-project",
+        dbt_project_path="/some/project/path",
+        **{project_config_arg: {"key": "value"}},  # type: ignore
+    )
+    execution_config = ExecutionConfig()
+    render_config = RenderConfig()
+    profile_config = MagicMock()
+    operator_args = {operator_arg: {"key": "value"}}
+
+    expected_error_msg = f"ProjectConfig.{project_config_arg} and operator_args with '{operator_arg}' are mutually exclusive and only one can be used."
+    with pytest.raises(CosmosValueError, match=expected_error_msg):
+        validate_initial_user_config(execution_config, profile_config, project_config, render_config, operator_args)
+
+
+def test_validate_user_config_fails_project_config_render_config_env_vars():
+    """
+    The validation should fail if a user specifies both ProjectConfig.env_vars and RenderConfig.env_vars.
+    """
+    project_config = ProjectConfig(env_vars={"key": "value"})
+    execution_config = ExecutionConfig()
+    render_config = RenderConfig(env_vars={"key": "value"})
+    profile_config = MagicMock()
+    operator_args = {}
+
+    expected_error_match = "Both ProjectConfig.env_vars and RenderConfig.env_vars were provided.*"
+    with pytest.raises(CosmosValueError, match=expected_error_match):
```

Comment:
This is very nice. I didn't know about the `match` argument! We can benefit from this in other tests.